### DISCUSSION TITLE
Make Orchid (more) International

### DISCUSSION THREAD

**Discussion by blankster** [orchidsoftware/platform#1545] [Ideas]
To spread Orchid over the global, localization is the key. Instructions and manuals are available in EN and RU. Users/Developers using Orchid already translated it to 10 languages as everybody can see  in https://github.com/orchidsoftware/platform/tree/master/resources/lang

Is your native language already available?

As I did the German translation, I know the translation to another language takes approx. 30 minutes. It would be great if your language isn't listed above if you could translate it to your native language too.

You must do a PR and name the file accordingly https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes

As more people using Orchid and translating/improving translations, as better it gets for everyone :) 

_If you need help, I'll do my best to provide basic support for translations._

**Comment by dhb52** [comment]
IMHO, the translation holds too many keys, including example screen. 
Maybe only the internal translation should be put into the package, while keep the example screen message English.  
So the translation work would be much easier, and the contributor can focus more on the quality of the translation work.
I think an artisan command is required to scan source code for `tran()/__()` and maintain the internal translation message, and meanwhile keep the keys sorted. When new keys added, the command line should also sync from the source code, and update all the existing json files by adding untranslated keys like 
```json
    "System": "__UNTRANSLATED__"
```

**Reply by tabuna** [reply]
I'm not sure if it's a good idea to add `__UNTRANSLATED__`. This can be easily seen by the user, instead of English (which is a fallback.)

To avoid this, every time we add at least one line with a translation, we will have to wait for its translations into all supported languages (Otherwise, the user will see `__UNTRANSLATED__`)

**Reply by tabuna** [reply]
The command to scan should be an internal command (only available to developers?). I've seen packages that do just that before. Maybe you should try them and if they suit us to write a tutorial, how to translate effectively?

**Reply by dhb52** [reply]
I add an PR #1547 for tooling

**Reply by blankster** [reply]
I add [here](https://github.com/dhb52/orchid-lang) a link to your repo to keep everything interlinked - thanks for your work!

**Reply by tabuna** [reply]
Hi @dhb52 , yes, great. I'll take a look and do a PR soon
